File: SayiAnalizi/SayiAnalizi.java
package sayianalizi;

import java.util.Scanner;

public class SayiAnalizi {

    public static void main(String[] args) {
        Scanner reader = new Scanner(System.in);
        System.out.print("Bir sayı giriniz: ");
        int sayi = reader.nextInt();
        int tekrakamlartoplami = 0;
        int ciftrakamlartoplami = 0;
        int tekrakamlarsayisi = 0;
        int ciftrakamlarsayisi = 0;
        while (sayi != 0) {
            int kalan = sayi % 10;
            sayi = sayi / 10;
            if (kalan % 2 == 0) {
                ciftrakamlartoplami = ciftrakamlartoplami + kalan;
                ciftrakamlarsayisi++;
            } else {
                tekrakamlartoplami = tekrakamlartoplami + kalan;
                tekrakamlarsayisi++;
            }
        }
        System.out.println("Basamak sayısı:" + (tekrakamlarsayisi + ciftrakamlarsayisi));
        System.out.println("Tek rakamların toplamı:" + tekrakamlartoplami);
        System.out.println("Çift rakamların toplamı:" + ciftrakamlartoplami);
        System.out.println("Tek ve çift rakamların farklarının mutlak değeri:" + Math.abs(tekrakamlartoplami - ciftrakamlartoplami));
    }
}

File: SayiAnalizi/SayiAnalizi.py
def sayi_analizi(sayi):

    basamak_sayisi = 0
    tek_rakamlarin_toplami = 0
    cift_rakamlarin_toplami = 0

    while sayi > 0:

        basamak = sayi % 10

        if basamak % 2 == 0:
            cift_rakamlarin_toplami += basamak
        else:
            tek_rakamlarin_toplami += basamak

        basamak_sayisi += 1
        sayi = int(sayi / 10)
    else:
        print("Basamak sayısı : {}".format(basamak_sayisi))
        print("Tek rakamların toplamı : {}".format(tek_rakamlarin_toplami))
        print("Çift rakamların toplamı : {}".format(cift_rakamlarin_toplami))
        print("Tek ve çift rakamların farklarının mutlak değeri : {}".format(abs(tek_rakamlarin_toplami-cift_rakamlarin_toplami)))


sayi_analizi(int(input("Bir sayı girin : ")))